test/asm/opt-r.asm

```diff
@@ -0,0 +1,13 @@
+OPT r34 ; :3
+OPT r 360
+
+; Invalid
+OPT r ; Missing arg
+OPT r 2a ; Bad decimal
+
+; Check that it has an effect
+OPT r 1
+MACRO m
+	m
```

Comment:
Can this not be an infinitely-recursive macro? IMO something which would *only* fail at `-r 1`, and be allowed for higher recursion levels, would be a more appropriate test case.

Reply:
(Before I noticed it's infinitely recursive, I was wondering whether we have an off-by-one error with how `-r` behaves. I'd expect `-r 0` to allow assembling a file without any macro expansions, `-r 1` to allow macro expansions but no macro usage within macro bodies, and so on.)

Reply:
This allows verifying at which recursion level the test is tripped, which is easier to debug.

Reply:
`-r 1` would currently fail with *any* macro expansion.

Reply:
How so? The error message would still be `Recursion limit (1) exceeded` even for:
```
MACRO m1
    m2
ENDM
MACRO m2
    println "m2"
ENDM
    m1
```

Reply:
And more specifically, it could test that `-r 1` does *not* fail for:
```
MACRO m0
    println "m0"
ENDM
    m0
```

Reply:
> `-r 1` would currently fail with any macro expansion.

Okay; that's a separate issue then.

Reply:
```console
% rgbasm -r 1 - <<EOF
MACRO m1
        print "hi"
ENDM
        m1
EOF
FATAL: <stdin>(4):
    Recursion limit (1) exceeded
```